beacon-chain/sync/subscriber_light_client.go

```diff
@@ -0,0 +1,66 @@
+package sync
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/OffchainLabs/prysm/v6/beacon-chain/core/feed"
+	statefeed "github.com/OffchainLabs/prysm/v6/beacon-chain/core/feed/state"
+	lightclientTypes "github.com/OffchainLabs/prysm/v6/consensus-types/light-client"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/proto"
+)
+
+func (s *Service) lightClientOptimisticUpdateSubscriber(_ context.Context, msg proto.Message) error {
```

Comment:
You are never calling this, or the finality subscriber, or the validation functions. You need to add https://github.com/OffchainLabs/prysm/pull/14983/files#diff-75e33ac18ab76e5853743737bca3d49b0d566959eb9b37381557a6a65f227c30 (plus gate it behind a flag per Nishant's comment)